degirum_tools/classification_eval.py

```diff
@@ -0,0 +1,152 @@
+import yaml, os
+from tqdm import tqdm
+from pathlib import Path
+from typing import Dict, List
+
+
+def model_prediction(model, validation_images_dir, label, foldermap, top_k=[1, 5]):
+    image_dir_path = Path(validation_images_dir + "/" + label)
+    image_extensions = {".jpg", ".jpeg", ".png", ".bmp", ".tif", ".tiff"}
+    all_images = [
+        str(image_path)
+        for image_path in image_dir_path.glob("*")
+        if image_path.suffix.lower() in image_extensions
+    ]
+    val_data = []
+    prediction: Dict[int, List[List[str]]] = {
+        k: [] for k in top_k
+    }  # Initialize prediction dictionary
+    for img in tqdm(all_images, desc="Processing Images"):
+        val_data.append(img)
+        with model:
+            res = model(img)
+            sorted_predictions = sorted(
+                res.results, key=lambda x: x["score"], reverse=True
+            )
+            for k in top_k:
+                top_predictions = sorted_predictions[:k]  # get the top k predictions
+                top_categories = [pred["category_id"] for pred in top_predictions]
+                prediction[k].append(
+                    [
+                        foldermap[int(top_categories[i])]
+                        for i in range(len(top_categories))
+                    ]
+                )
+    return (prediction, val_data)
+
+
+def identify_misclassified_examples(prediction, gndtruth, val_data):
+    misclassified_images: Dict[int, List[List[str]]] = {
+        k: [] for k in prediction.keys()
+    }  # Initialize misclassified_images dictionary
+
+    for k, preds in prediction.items():  # Iterate over each top-k prediction
+        for i in range(len(preds)):  # Iterate over predictions for each image
+            if (
+                gndtruth[i] not in preds[i]
+            ):  # Check if ground truth is in top-k predictions
+                misclassified_images[k].append(
+                    val_data[i]
+                )  # Append misclassified image to the corresponding list
+        print(
+            f"Count of misclassified examples for Top-{k}: {len(misclassified_images[k])}\n"
+        )
+
+
+def calculate_accuracy(predictions, ground_truth, top_k):
```

Comment:
can we calculate accuracy while running the model similar to Ultralytics.
So from the beginning we know if there is something off.
If we use tqdm with predict_batch there should be a progress bar and show the accuracy live.